main/src/ca/uwaterloo/flix/language/phase/typer/RecordConstraintSolver2.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.typer
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.shared.Scope
+import ca.uwaterloo.flix.language.ast.{Kind, Name, RigidityEnv, Symbol, Type, TypeConstructor}
+import ca.uwaterloo.flix.language.phase.typer.ConstraintSolver2.{Progress, TypeConstraint}
+import ca.uwaterloo.flix.language.phase.unification.Substitution
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+object RecordConstraintSolver2 {
+
+  /**
+    * Unifies the two given record row types.
+    */
+  def solve(tpe1: Type, tpe2: Type)(implicit scope: Scope, tracker: Progress, renv: RigidityEnv, flix: Flix): (List[TypeConstraint], Substitution) = (tpe1, tpe2) match {
```

Comment:
I would consider putting the tracker as the last argument just before flix.

Reply:
What do you think of making scope and renv explicit arguments? It seems they are *actual* arguments to the function.

Reply:
tracker: Progress -- So which is it? :)